docs/decisions/0009-moving-catalog-diffing-to-enterprise-catalog.rst

```diff
@@ -0,0 +1,89 @@
+scheduled run, we cannot accurately determine if all content under a customer's catalog has been updated since the
+catalog has last updated without first using the enterprise catalog's currently available API `get_content_metadata`
+endpoint.
+
+Consequences
+============
+
+**querying all content from the catalog service is expensive**- Asking for all content under the catalog, for every
+customer, for every catalog, on every transmission is not only an undesirably large request to be sending, but also
+consumes much of the enterprise catalog's DB load. We need a smarter solution to how the Integrated Channels requests
+content from the catalog service.
+
+**putting the responsibility of determining differences** between what's already been sent to the customers and what's
+currently in customer's catalogs adds much complexity to the integrated channels which can make it unapproachable from
+an onboarding perspective but also from a debugging point of view. Log noise and number of potential error sources means
+tracking down issues within the Integrated Channels are costly and time consuming.
+
+How do the Integrated Channels currently handle customer catalog data?
+======================================================================
+
+existing flow chart:
+
+.. image:: ../images/0009-moving-catalog-diffing-to-enterprise-catalog.rst
+  :width: 500
+
+Currently on a scheduled job, the Integrated Channels' content metadata exporter iterates over each customer's catalog
+and builds up a dict of content representing the current state of the customer's catalog content. Take away- as it
+stands now, the Integrated Channels assumes that the content metadata exporter will report the entirety of the
+customer's purchased content, then pass that payload to the content metadata transmitter which compares the payload with
+all saved `ContentMetadataItemTransmission` items under the customer to determine which items it (the transmitter) needs
+to create, update, and delete. It then chunks up each of the buckets, and sends those chunks (up to a limited number of
+times) to the client to send to the customer's LMS and saves the updated `ContentMetadataItemTransmission` entry.
+
+How the Integrated Channels could better determine if customer metadata needs updating
+======================================================================================
+
+proposed flow chart:
+
+.. image:: ../images/new-integrated-channels-metadata-flow.png
+  :width: 500
+
+If we build an endpoint that takes catalog UUIDs and a set of content keys linked to the uuid and returns three buckets
+of data:
+
+1) What content keys exist under the specified catalog that were not provided in the list of content keys
+
+2) What content keys don't exist under the specified catalog that were provided in the list of content keys
+
+3) What the last updated at times of the content keys are for the content keys provided that are under the specified
```

Comment:
This is the primary use case for the endpoint so I think we should list it first versus third.